app/src/main/java/com/practicum/playlistmaker/SettingsActivity.kt

```diff
@@ -18,10 +22,33 @@ class SettingsActivity : AppCompatActivity() {
             insets
         }
 
-        val buttonBack = findViewById<Button>(R.id.btn_back)
-
-        buttonBack.setOnClickListener {
+        val buttonBack = findViewById<MaterialToolbar>(R.id.btn_back)
+        buttonBack.setNavigationOnClickListener {
             finish()
         }
+
+        val buttonShare = findViewById<Button>(R.id.btn_share)
+        buttonShare.setOnClickListener {
+            val intent = Intent(Intent.ACTION_SEND)
+            intent.type = "text/plain"
+            intent.putExtra(Intent.EXTRA_TEXT, getString(R.string.link_course))
+            startActivity(intent)
+        }
+
+        val buttonSupport = findViewById<Button>(R.id.btn_support)
+        buttonSupport.setOnClickListener {
+            val intent = Intent(Intent.ACTION_SENDTO)
+            intent.data = "mailto:".toUri()
+            intent.putExtra(Intent.EXTRA_EMAIL, arrayOf(R.string.email))
```

Comment:
Здесь нужно вот так сделать, чтобы поле "Кому" предзаполнялось корректно:
```suggestion
            intent.putExtra(Intent.EXTRA_EMAIL, arrayOf(getString(R.string.email)))
```